vmdb/app/models/ems_refresh/save_inventory.rb

```diff
@@ -173,6 +168,10 @@ def save_guest_devices_inventory(hardware, hashes)
         # Save the hardware to force an id if not found
         hardware.save! if hardware.id.nil?
         h[:network][:hardware_id] = hardware.id
+      else
+        # ensure save_network_inventory gets called
+        # it will be deleted
+        h[:network] = {}
```

Comment:
@Fryguy corrected me here.
It calls into the children if the key is present. So this is not necessary.

The issue of nil not properly saving an item is still an issue. Will address in a different PR.
Looks like the return if `hashes.nil?` is here to stay for a while. (that was my main goal)